main.py

```diff
@@ -108,9 +108,11 @@ def main():
             elif choice == "2":  # Edit Grade
                 clear_screen()
                 print("========Edit Grade========")
-                student_id = int(input("Enter Student ID: "))
-                new_grade = input("Enter New Grade: ")
-                gradebook.edit_grade(instructor, course_id, student_id, new_grade)
+                grade_exists = gradebook.grades_to_edit(instructor, course_id)
+                if grade_exists == True:
+                    student_id = int(input("Enter Student ID: "))
+                    new_grade = input("Enter New Grade: ")
```

Comment:
Please add appropriate try/catch blocks to handle cases where users enter invalid student IDs or grades. Please implement a method to validate student_id and another to validate grades (one of them might already exist).

Place these validation methods in util.py, and then make sure to use them wherever student IDs or grades are processed.